### ISSUE TITLE
npm run build fails unable to find "path", "fs" and others

### ISSUE THREAD

**Issue by Sanscripter** [angular/angular#54235]
### Which @angular/* package(s) are the source of the bug?

Don't known / other

### Is this a regression?

Yes

### Description

I'm unable to build my project (linked as reproduction, just clone and npm run build)  due to system unable to find modules baked into node. Found at least one other reference to this issue online: https://stackoverflow.com/questions/77918038/problem-while-ng-build-using-the-new-angular-17-application-builder though also unanswered.

### Please provide a link to a minimal reproduction of the bug

https://github.com/Sanscripter/trading-grounds

### Please provide the exception or error you saw

```true
X [ERROR] Could not resolve "path"

    (disabled):node_modules/jsdom/lib/api.js:3:21:
      3 │ const path = require("path");
        ╵                      ~~~~~~

  The package "path" wasn't found on the file system but is built into node. Are you trying to bundle for node? You can use "platform: 'node'" to do that, which will remove this error.


X [ERROR] Could not resolve "fs"

    (disabled):node_modules/jsdom/lib/api.js:4:19:
      4 │ const fs = require("fs").promises;
        ╵                    ~~~~

  The package "fs" wasn't found on the file system but is built into node. Are you trying to bundle for node? You can use "platform: 'node'" to do that, which will remove this error.


X [ERROR] Could not resolve "path"

    (disabled):node_modules/jsdom/lib/jsdom/utils.js:3:21:
      3 │ const path = require("path");
        ╵                      ~~~~~~

  The package "path" wasn't found on the file system but is built into node. Are you trying to bundle for node? You can use "platform: 'node'" to do that, which will remove this error.


X [ERROR] Could not resolve "fs"

    node_modules/jsdom/lib/jsdom/browser/resources/resource-loader.js:3:19:
      3 │ const fs = require("fs");
        ╵                    ~~~~

  The package "fs" wasn't found on the file system but is built into node. Are you trying to bundle for node? You can use "platform: 'node'" to do that, which will remove this error.


X [ERROR] Could not resolve "url"

    node_modules/jsdom/lib/jsdom/browser/resources/resource-loader.js:6:12:
      6 │ } = require("url");
        ╵             ~~~~~

  The package "url" wasn't found on the file system but is built into node. Are you trying to bundle for node? You can use "platform: 'node'" to do that, which will remove this error.


X [ERROR] Could not resolve "http"

    node_modules/jsdom/lib/jsdom/living/helpers/agent-factory.js:3:21:
      3 │ const http = require("http");
        ╵                      ~~~~~~

  The package "http" wasn't found on the file system but is built into node. Are you trying to bundle for node? You can use "platform: 'node'" to do that, which will remove this error.


X [ERROR] Could not resolve "https"

    node_modules/jsdom/lib/jsdom/living/helpers/agent-factory.js:4:22:
      4 │ const https = require("https");
        ╵                       ~~~~~~~

  The package "https" wasn't found on the file system but is built into node. Are you trying to bundle for node? You can use "platform: 'node'" to do that, which will remove this error.


X [ERROR] Could not resolve "url"

    node_modules/jsdom/lib/jsdom/living/helpers/agent-factory.js:7:12:
      7 │ } = require("url");
        ╵             ~~~~~

  The package "url" wasn't found on the file system but is built into node. Are you trying to bundle for node? You can use "platform: 'node'" to do that, which will remove this error.


X [ERROR] Could not resolve "http"

    node_modules/jsdom/lib/jsdom/living/helpers/http-request.js:3:21:
      3 │ const http = require("http");
        ╵                      ~~~~~~

  The package "http" wasn't found on the file system but is built into node. Are you trying to bundle for node? You can use "platform: 'node'" to do that, which will remove this error.


X [ERROR] Could not resolve "https"

    node_modules/jsdom/lib/jsdom/living/helpers/http-request.js:4:22:
      4 │ const https = require("https");
        ╵                       ~~~~~~~

  The package "https" wasn't found on the file system but is built into node. Are you trying to bundle for node? You can use "platform: 'node'" to do that, which will remove this error.


X [ERROR] Could not resolve "stream"

    node_modules/jsdom/lib/jsdom/living/helpers/http-request.js:7:12:
      7 │ } = require("stream");
        ╵             ~~~~~~~~

  The package "stream" wasn't found on the file system but is built into node. Are you trying to bundle for node? You can use 
"platform: 'node'" to do that, which will remove this error.


X [ERROR] Could not resolve "zlib"

    node_modules/jsdom/lib/jsdom/living/helpers/http-request.js:8:21:
      8 │ const zlib = require("zlib");
        ╵                      ~~~~~~

  The package "zlib" wasn't found on the file system but is built into node. Are you trying to bundle for node? You can use "platform: 'node'" to do that, which will remove this error.


X [ERROR] Could not resolve "util"

    node_modules/jsdom/lib/jsdom/living/helpers/runtime-script-errors.js:3:21:
      3 │ const util = require("util");
        ╵                      ~~~~~~

  The package "util" wasn't found on the file system but is built into node. Are you trying to bundle for node? You can use "platform: 'node'" to do that, which will remove this error.


X [ERROR] Could not resolve "os"

    node_modules/jsdom/lib/jsdom/living/navigator/NavigatorConcurrentHardware-impl.js:3:19:
      3 │ const os = require("os");
        ╵                    ~~~~

  The package "os" wasn't found on the file system but is built into node. Are you trying to bundle for node? You can use "platform: 'node'" to do that, which will remove this error.


X [ERROR] Could not resolve "http"

    node_modules/jsdom/lib/jsdom/living/xhr/XMLHttpRequest-impl.js:3:34:
      3 │ const HTTP_STATUS_CODES = require("http").STATUS_CODES;
        ╵                                   ~~~~~~

  The package "http" wasn't found on the file system but is built into node. Are you trying to bundle for node? You can use "platform: 'node'" to do that, which will remove this error.


X [ERROR] Could not resolve "child_process"

    node_modules/jsdom/lib/jsdom/living/xhr/XMLHttpRequest-impl.js:6:12:
      6 │ } = require("child_process");
        ╵             ~~~~~~~~~~~~~~~

  The package "child_process" wasn't found on the file system but is built into node. Are you trying to bundle for node? You can use "platform: 'node'" to do that, which will remove this error.


X [ERROR] Could not resolve "fs"

    node_modules/jsdom/lib/jsdom/living/xhr/xhr-utils.js:3:19:
      3 │ const fs = require("fs");
        ╵                    ~~~~

  The package "fs" wasn't found on the file system but is built into node. Are you trying to bundle for node? You can use "platform: 'node'" to do that, which will remove this error.


X [ERROR] Could not resolve "path"

    node_modules/jsdom/lib/jsdom/utils.js:3:21:
      3 │ const path = require("path");
        ╵                      ~~~~~~

  The package "path" wasn't found on the file system but is built into node. Are you trying to bundle for node? You can use "platform: 'node'" to do that, which will remove this error.


X [ERROR] Could not resolve "net"

    node_modules/jsdom/node_modules/http-proxy-agent/dist/agent.js:38:38:
      38 │ const net_1 = __importDefault(require("net"));
         ╵                                       ~~~~~

  The package "net" wasn't found on the file system but is built into node. Are you trying to bundle for node? You can use "platform: 'node'" to do that, which will remove this error.


X [ERROR] Could not resolve "tls"

    node_modules/jsdom/node_modules/http-proxy-agent/dist/agent.js:39:38:
      39 │ const tls_1 = __importDefault(require("tls"));
         ╵                                       ~~~~~

  The package "tls" wasn't found on the file system but is built into node. Are you trying to bundle for node? You can use "platform: 'node'" to do that, which will remove this error.


X [ERROR] Could not resolve "url"

    node_modules/jsdom/node_modules/http-proxy-agent/dist/agent.js:40:38:
      40 │ const url_1 = __importDefault(require("url"));
         ╵                                       ~~~~~

  The package "url" wasn't found on the file system but is built into node. Are you trying to bundle for node? You can use "platform: 'node'" to do that, which will remove this error.


X [ERROR] Could not resolve "net"

    node_modules/jsdom/node_modules/https-proxy-agent/dist/agent.js:38:38:
      38 │ const net_1 = __importDefault(require("net"));
         ╵                                       ~~~~~

  The package "net" wasn't found on the file system but is built into node. Are you trying to bundle for node? You can use "platform: 'node'" to do that, which will remove this error.


X [ERROR] Could not resolve "tls"

    node_modules/jsdom/node_modules/https-proxy-agent/dist/agent.js:39:38:
      39 │ const tls_1 = __importDefault(require("tls"));
         ╵                                       ~~~~~

  The package "tls" wasn't found on the file system but is built into node. Are you trying to bundle for node? You can use "platform: 'node'" to do that, which will remove this error.


X [ERROR] Could not resolve "url"

    node_modules/jsdom/node_modules/https-proxy-agent/dist/agent.js:40:38:
      40 │ const url_1 = __importDefault(require("url"));
         ╵                                       ~~~~~

  The package "url" wasn't found on the file system but is built into node. Are you trying to bundle for node? You can use "platform: 'node'" to do that, which will remove this error.

X [ERROR] Could not resolve "assert"

    node_modules/jsdom/node_modules/https-proxy-agent/dist/agent.js:41:41:
      41 │ const assert_1 = __importDefault(require("assert"));
         ╵                                          ~~~~~~~~

  The package "assert" wasn't found on the file system but is built into node. Are you trying to bundle for node? You can use 
"platform: 'node'" to do that, which will remove this error.

Application bundle generation failed. [10.511 seconds]
```

### Please provide the environment you discovered this bug in (run `ng version`)

```true
Angular CLI: 17.0.10
Node: 18.13.0
Package Manager: npm 8.19.3
OS: win32 x64

Angular: 17.0.9
... animations, common, compiler, compiler-cli, core, forms
... platform-browser, platform-browser-dynamic, router

Package                         Version
---------------------------------------------------------
@angular-devkit/architect       0.1700.10
@angular-devkit/build-angular   17.0.10
@angular-devkit/core            17.0.10
@angular-devkit/schematics      17.0.10
@angular/cli                    17.0.10
@schematics/angular             17.0.10
rxjs                            7.8.1
typescript                      5.2.2
zone.js                         0.14.3
```

### Anything else?

Not sure if it matters but I use NVM (for windows) to manage my node version.

**Comment by giuseppebrb** [comment]
I was facing the same issue with the same packages (migrated an existing angular v16 app to v17 and changed the builder from webpack to application). I was struggling but at the end I discovered that the problem was a dependency which, after the change of the building system, needed the browser-compatible version.
So in the `package.json` I changed `"fabric": "^5.2.1",` to `"fabric": "^5.3.0-browser",` which is the same library but browser compatible. 
Check your dependencies and switch to the browser versions
